sql/analyzer/rules.go

```diff
@@ -531,14 +534,82 @@ func resolveNaturalJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, e
 
 		projections := append(append(common, left...), right...)
 
-		return plan.NewProject(
+		transformed = plan.NewProject(
 			projections,
 			plan.NewInnerJoin(
 				join.Left,
 				join.Right,
 				expression.JoinAnd(conditions...),
 			),
-		), nil
+		)
+
+		return transformed, nil
+	})
+
+	if err != nil || transformed == nil {
+		return node, err
+	}
+
+	var nodesToUnresolve = []sql.Node{}
+	plan.Inspect(node, func(iNode sql.Node) bool {
```

Comment:
why `iNode`?

Reply:
to avoid use n and node so many times in different context, but yes it looks weird (stands for `inspected node`). Changing it